### PR TITLE
fix: debounce for search queries

### PR DESCRIPTION
**Pull request by tm-ruxandra**
## Description

Wrap the map query suggestion system in a 500ms debounce to avoid hammering the map or related downstream APIs. Also refactor the component to extract the query suggestion system to a self-contained hook which the component uses. Additionally, wrap relevant functions in `useCallback` and handle `zoomTo` calls through a single side-effect driven by the suggestion system's selected coordinates rather than scattered through async functions. We believe that the hang issue was caused by some combination of these factors (many async calls affecting zoom position) and that this will mitigate the issue, but were not able to find a direct repro case.

### Checklist
- [x] Corresponding issue has been opened

### Related Issues
#1827 

### Verification steps

Enter string and coordinate-based queries in the sites screen search bar and verify that behavior is still as expected, with the addition of a 500ms debounce window before input results in on-screen changes.

### CODE REVIEW

File: dev-client/src/hooks/useMapSuggestions.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import {useCallback, useState} from 'react';
```
Comment (paulschreiber) [inline dev-client/src/hooks/useMapSuggestions.ts:1]: issue: add copyright notice.

File: dev-client/src/hooks/useMapSuggestions.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      newAbortController.signal,
+    );
+    setAbortController(null);
+    return newSuggestions;
+  }, []);
+
+  const querySuggestions = useCallback(
+    async (queryText: string) => {
+      if (isValidCoordinates(queryText)) {
+        const [latitude, longitude] = queryText.split(',').map(Number);
```
Comment (paulschreiber) [inline dev-client/src/hooks/useMapSuggestions.ts:50]: suggestion: validate the coordinates before submitting them. (we have `isValidLatitude`/ `isValidLongitude` helpers already).

but: not sure what to do if they're invalid. Ask @CourtneyLee333?

This also is somewhat out of scope, so we could do that work in another PR.

File: dev-client/src/hooks/useMapSuggestions.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            }
+            return null;
+          });
+          setSuggestions([]);
+        }
+      }
+    },
+    [makeSuggestionsApiCall],
+  );
+  const debouncedQuerySuggestions = useDebouncedCallback(querySuggestions, 500);
```
Comment (paulschreiber) [inline dev-client/src/hooks/useMapSuggestions.ts:78]: suggestion: put the debounce time in a constant.

File: dev-client/src/hooks/useMapSuggestions.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+import {MAP_QUERY_MIN_LENGTH} from 'terraso-mobile-client/constants';
+import {
+  initMapSearch,
+  Suggestion,
+} from 'terraso-mobile-client/services/mapSearchService';
+import {isValidCoordinates} from 'terraso-mobile-client/util';
+
+/*
+ * Logic extracted from screens/HomeScreen/components/MapSearch.tsx; see its revisions
```
Comment (knipec) [inline dev-client/src/hooks/useMapSuggestions.ts:15]: Oo I haven't been doing this when extracting things into new files but I like it

File: dev-client/src/hooks/useMapSuggestions.ts (hunk 2 above)
Comment (tm-ruxandra) [reply dev-client/src/hooks/useMapSuggestions.ts:50]: I don't understand your suggestion - we're invoking `isValidCoordinates` right here which in turn invokes `isValidLatitude(latitude) && isValidLongitude(longitude)`.

**Review by knipec — APPROVED** [review]

File: dev-client/src/hooks/useMapSuggestions.ts (hunk 2 above)
Comment (paulschreiber) [reply dev-client/src/hooks/useMapSuggestions.ts:70]: Oops.
